pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -197,50 +212,59 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
-		reqLogger.Info("Creating a new Deployment", "Deployment.Namespace", postgresqlDeployment.Namespace, "Deployment.Name", postgresqlDeployment.Name)
-		err = r.client.Create(context.TODO(), postgresqlDeployment)
+		reqLogger.Info("Creating a new ImageStream", "ImageStream.Namespace", postgresImageStream.Namespace, "ImageStream.Name", postgresImageStream.Name)
+		err = r.client.Create(context.TODO(), postgresImageStream)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 	} else if err != nil {
 		return reconcile.Result{}, err
-	} else {
-		desiredImage := postgresql.image()
+	}
+	//#endregion
 
-		containerSpec := findContainerSpec(foundPostgresqlDeployment, POSTGRES_CONTAINER_NAME)
-		if containerSpec == nil {
-			reqLogger.Info("Skipping image reconcile: Unable to find container spec in deployment", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "ContainerSpec", POSTGRES_CONTAINER_NAME)
-		} else if containerSpec.Image != desiredImage {
-			reqLogger.Info("Container spec in deployment is using a different image. Going to update it now.", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "ContainerSpec", POSTGRES_CONTAINER_NAME, "ExistingImage", containerSpec.Image, "DesiredImage", desiredImage)
+	//#region Postgres DeploymentConfig
+	postgresqlDeploymentConfig, err := newPostgresqlDeploymentConfig(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 
-			// update
-			updateContainerSpecImage(foundPostgresqlDeployment, POSTGRES_CONTAINER_NAME, desiredImage)
+	// Set UnifiedPushServer instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, postgresqlDeploymentConfig, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
 
-			// enqueue
-			err = r.client.Update(context.TODO(), foundPostgresqlDeployment)
-			if err != nil {
-				reqLogger.Error(err, "Failed to update Deployment", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name)
-				return reconcile.Result{}, err
-			}
-			return reconcile.Result{Requeue: true}, nil
+	// Check if this DeploymentConfig already exists
+	foundPostgresqlDeploymentConfig := &openshiftappsv1.DeploymentConfig{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: postgresqlDeploymentConfig.Name, Namespace: postgresqlDeploymentConfig.Namespace}, foundPostgresqlDeploymentConfig)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating a new DeploymentConfig", "DeploymentConfig.Namespace", postgresqlDeploymentConfig.Namespace, "DeploymentConfig.Name", postgresqlDeploymentConfig.Name)
+		err = r.client.Create(context.TODO(), postgresqlDeploymentConfig)
+		if err != nil {
+			return reconcile.Result{}, err
 		}
+	} else if err != nil {
+		return reconcile.Result{}, err
 	}
+	//#endregion
 
+	//#region Postgres Service
```

Comment:
These region blocks look really good on Intellij and VSCode (with extension).
They're also code regular comments specifying the block start and end.

![Screenshot from 2019-06-21 15-07-48](https://user-images.githubusercontent.com/376732/59925761-79dfb580-9441-11e9-8881-1b9a52cb41e3.png)

Reply:
@grdryn don't know about Emacs though!

Reply:
Cool! :+1: